docs/demo.md

```diff
@@ -0,0 +1,227 @@
+
+Th above steps show how Kai simplifies the translation of import namespaces, ensuring seamless automated migration of javax libraries to jakarta persistence libraries.
+
+#### 2.2.2 Modify Scope from CDI bean requirements
+
+In this step, we will use Kai to modify the scope in `CatalogService.java` to adhere to Quarkus CDI bean requirements. Kai will handle this automatically, ensuring seamless migration.
+
+- Right-click on the file `src/main/java/com/redhat/coolstore/service/CatalogService.java`.
+- Select Kai Fix-All.
+- Accept the proposed changes.
+
+![Modify Scope from CDI bean requirements](kai_type2.png)
+
+#### 2.2.3 EJB Remote and Message Driven Bean(MDB) changes
+
+We will address EJB Remote and MDB functionalities in `ShippingService.java` and `ShoppingCartOrderProcessor.java` respectively. Kai will guide us through replacing EJBs with REST functionality and updating related imports and annotations.
+
+##### EJB Remote
+
+- Right-click on the file `src/main/java/com/redhat/coolstore/service/ShippingService.java`.
+- Select `Kai Fix-All`.
+- Accept the proposed changes.
+
+After applying the changes, you may notice unwanted imports such as `import jakarta.ejb.Singleton;`. Remove this import and save the file.
+
+![EJB Remote - Before/After](ejb_remote.png)
+
+Due to the absence of support for Remote EJBs in Quarkus, you will notice that these functionalities are removed and replaced with REST functionality.
+
+##### Message Driven Bean(MDB)
+
+- Right-click on the file `src/main/java/com/redhat/coolstore/service/ShoppingCartOrderProcessor.java`.
+
+- Select `Kai Fix-All` and accept the changes.
+
+- You will notice unwanted imports like `import jakarta.ejb.Singleton;`. Remove this import from the file and save it.
+
+- Check for any missing imports. If you find `import jakarta.enterprise.context.ApplicationScoped;` missing, add it to the file.
+
+![MDB - before/after](mdb.png)
+
+- Note the changes made to `ordersEmitter` channel:
+
+  ```java
+  @Inject
+  @Channel("orders")
+  Emitter<String> ordersEmitter;
+  ```
+
+  Although this change is valid, it's important to note that the LLM is unaware of the number of consumers for `orders` channel. It assumed that there is a single consumer for `orders`, but, this app has multiple consumers of `orders`. Assuming there's only a single consumer can lead to `io.smallrye.reactive.messaging.providers.wiring.TooManyDownstreamCandidatesException`. To avoid this, add the `@Broadcast` annotation:
+
+  ```java
+  import io.smallrye.reactive.messaging.annotations.Broadcast;
+
+  @Inject
+  @Channel("orders")
+  @Broadcast
+  Emitter<String> ordersEmitter;
+  ```
+
+- Since the LLM doesn't have access to the project's `pom.xml` file, it lacks information about available dependencies. Update the import statements to the following:
+
+  ```java
+  import org.eclipse.microprofile.reactive.messaging.Channel;
+  import org.eclipse.microprofile.reactive.messaging.Emitter;
+  ```
```

Comment:
Using cached data I saw these 2 imports were already added

<img width="1718" alt="Screenshot 2024-05-21 at 1 48 24 PM" src="https://github.com/konveyor-ecosystem/kai/assets/206875/cddfeb18-db1d-43b9-b525-c55821598d02">

Reply:
I'm mistaken, the import from cached data was:
```
import io.smallrye.reactive.messaging.Channel;
import io.smallrye.reactive.messaging.Emitter;
```

and this doc is recommending:
```
  import org.eclipse.microprofile.reactive.messaging.Channel;
  import org.eclipse.microprofile.reactive.messaging.Emitter;

```

So I still need to make a change as per doc

Reply:
ack